content/blog/switching-the-application-stack-from-pern-to-mern/index.md

```diff
@@ -0,0 +1,392 @@
+            }
+        },
+        strategy: {
+            type: "Recreate"
+        }
+    }}, {
+    deleteBeforeReplace: true,
+    provider: eksCluster.provider,
+});
+
+const databasesideListener = new k8s.core.v1.Service("database-side-listener", {
+    metadata: { labels: databaseDeployment.metadata.labels },
+    spec: {
+        type: "ClusterIP",
+        ports: [{ port: 27017, targetPort: "http" }],
+        selector: databaseAppLabels,
+        publishNotReadyAddresses: false,
+    }}, {
+    provider: eksCluster.provider,
+    }
+);
+
+const mongodbAddress = databasesideListener.spec.clusterIP;
+
+const serverAppName = "server";
+const serverAppLabels = { appClass: serverAppName };
+const serverDeployment = new k8s.apps.v1.Deployment("server-side-service", {
+    metadata: { labels: serverAppLabels },
+    spec: {
+        replicas: 2,
+        selector: { matchLabels: serverAppLabels },
+        template: {
+            metadata: { labels: serverAppLabels },
+            spec: {
+                containers: [{
+                    name: serverAppName,
+                    image: awsx.ecr.buildAndPushImage("server-side-service", "./serverside").image(),
+                    ports: [{ name: "http", containerPort: 5000 }],
+                    env: [
+                        { name: "USER_NAME", value: sqlUserName },
+                        { name: "USER_PASSWORD", value: sqlUserPassword },
+                        { name: "MONGODB_ADDRESS", value: mongodbAddress },
+                        { name: "MONGODB_PORT", value: String(27017) },
+                        { name: "DATABASE_NAME", value: databaseName },
+                    ],
+                    resources: {
+                        limits: {
+                            memory: "500Mi",
+                            cpu: "500m"
+                        }
+                    }
+                }],
+            }
+        }
+    }}, {
+    provider: eksCluster.provider,
+});
+
+const serversideListener = new k8s.core.v1.Service(...);
+
+const clientAppName = ...;
+const clientAppLabels = { ... };
+const clientDeployment = new k8s.apps.v1.Deployment(...);
+
+const clientsideListener = new k8s.core.v1.Service(...);
+```
+
+To give us a simple way to connect to our application, we export the clientside listener's address. We can now open a browser window with the URL and port number to view our application.
+
+```typescript
+export const kubeConfig = eksCluster.kubeconfig;
+export const URL = clientsideListener.status.loadBalancer.ingress[0].hostname;
+```
+
+In this example, we showed how to change a PERN stack based application to use MERN, and demonstrated the ease with which a designer can swap out different components within their application. The isolated nature of containerized applications allows for a great deal of modularity, and can significantly reduce the amount of time spent renovating previously written code to fit any new components that are added.
```

Comment:
```suggestion
In this example, we showed how to change a PERN stack-based application to use MERN, and demonstrated the ease with which a designer can swap out different components. The isolated nature of containerized applications allows for a great deal of modularity. It can significantly reduce the amount of time spent renovating previously written code to fit any added new components.
```